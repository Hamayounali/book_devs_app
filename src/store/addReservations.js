import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const url = 'http://localhost:3000/api/v1/reservations';
const ADD_RESERVATION = 'book_devs/ADD_RESERVATION';

export const addReservation = createAsyncThunk(ADD_RESERVATION, async (
  { name, reservation_date, city, user_id, developer_id },
  thunkAPI,
) => {
  const data = {
    name, reservation_date, city, user_id, developer_id,
  };

  await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
    
});


const storeSlice = createSlice({
  name: 'book_devs',
  initialState: [],
  extraReducers: {
    [addReservation.fulfilled]: (state, action) => action.payload,
  },
});

// export const reservations = (state) => state.reservations;
export default storeSlice.reducer;
