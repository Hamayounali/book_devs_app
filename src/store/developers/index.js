export { getDevelopers, addDeveloper } from './devs/devActions';
