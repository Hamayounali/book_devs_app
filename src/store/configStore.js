import { combineReducers, configureStore } from '@reduxjs/toolkit';
import reservationsReducer from './reservations/reservations';
import authReducer from './user/auth';
import devReducer from './developers/devs/devReducer';
import addReservations from './reservations/addReservations';

const rootReducer = combineReducers({
  reservations: reservationsReducer,
  authentication: authReducer,
  developer: devReducer,
  addReservations,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
