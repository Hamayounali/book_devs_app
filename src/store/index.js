import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import addReservations from './addReservations';

const reducer = combineReducers({
  addReservations,
});

const store = configureStore({ reducer }, applyMiddleware(thunk));

export default store;
