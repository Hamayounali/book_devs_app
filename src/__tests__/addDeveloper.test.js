import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddDeveloper from '../components/devs/addDeveloper';
import store from '../store/configStore';

describe('Test Register page', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <AddDeveloper />
      </BrowserRouter>
    </Provider>,
  );
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Add Developer page has button Add Developer', () => {
    expect(tree.findByText('Add Developer')).toBeTruthy();
  });
});
