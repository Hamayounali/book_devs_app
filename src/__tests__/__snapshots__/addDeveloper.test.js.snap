// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Register page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-box"
      >
        <form>
          <div
            class="user-box"
          >
            <input
              class="form-control"
              id="name"
              required=""
              type="text"
              value=""
            />
            <label
              for="name"
            >
              name
            </label>
          </div>
          <div
            class="user-box"
          >
            <input
              class="form-control"
              id="photo"
              multiple=""
              required=""
              type="text"
            />
            <label
              for="photo"
            >
              Image
            </label>
          </div>
          <div
            class="user-box"
          >
            <input
              class="form-control"
              id="title"
              required=""
              type="text"
            />
            <label
              for="title"
            >
              title 
            </label>
          </div>
          <div
            class="user-box"
          >
            <input
              class="form-control"
              id="description"
              required=""
              type="text"
            />
            <label
              for="description"
            >
              Description
            </label>
          </div>
          <div
            class="user-box"
          >
            <input
              class="form-control"
              id="salary"
              required=""
              type="number"
            />
            <label
              for="salary"
            >
              salary
            </label>
          </div>
          <div
            class="user-box"
          >
            <input
              class="form-control"
              id="rating"
              required=""
              type="number"
            />
            <label
              for="rating"
            >
              rating
            </label>
          </div>
          <a
            href="#"
          >
            <span />
            <span />
            <span />
            <span />
            <button
              type="submit"
            >
              Add Developer
            </button>
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-box"
    >
      <form>
        <div
          class="user-box"
        >
          <input
            class="form-control"
            id="name"
            required=""
            type="text"
            value=""
          />
          <label
            for="name"
          >
            name
          </label>
        </div>
        <div
          class="user-box"
        >
          <input
            class="form-control"
            id="photo"
            multiple=""
            required=""
            type="text"
          />
          <label
            for="photo"
          >
            Image
          </label>
        </div>
        <div
          class="user-box"
        >
          <input
            class="form-control"
            id="title"
            required=""
            type="text"
          />
          <label
            for="title"
          >
            title 
          </label>
        </div>
        <div
          class="user-box"
        >
          <input
            class="form-control"
            id="description"
            required=""
            type="text"
          />
          <label
            for="description"
          >
            Description
          </label>
        </div>
        <div
          class="user-box"
        >
          <input
            class="form-control"
            id="salary"
            required=""
            type="number"
          />
          <label
            for="salary"
          >
            salary
          </label>
        </div>
        <div
          class="user-box"
        >
          <input
            class="form-control"
            id="rating"
            required=""
            type="number"
          />
          <label
            for="rating"
          >
            rating
          </label>
        </div>
        <a
          href="#"
        >
          <span />
          <span />
          <span />
          <span />
          <button
            type="submit"
          >
            Add Developer
          </button>
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
